### ISSUE TITLE
Null result not longer supported in nnbd version

### ISSUE THREAD

**Issue by alextekartik** [dart-lang/tools#735]
Thanks for this convenient implementation and for porting to null_safety. It seems you made the choice to no longer support null result:

https://github.com/dart-lang/json_rpc_2/blob/62fd274acd6ab93ed58cbf963c3153ddf8daa862/lib/src/server.dart#L204

However I don't see in the json-prc spec that result can not be null (the field is mandatory but its value can be null). Maybe I'm wrong but it definitely breaks some code I have so far. And well, I'm not sure what to return if I don't want to return anything (i.e. like I would do for a void function).

Basically just changing `Object result` to `Object? result` fixes the issue for me and I was about to propose a PR but I saw the puzzling multiple requests handling:

https://github.com/dart-lang/json_rpc_2/blob/62fd274acd6ab93ed58cbf963c3153ddf8daa862/lib/src/server.dart#L182-L183

that is removing null results (and there should not be any) so I'm not sure what should be done. I can work on a PR if needed and if a null result should indeed be allowed.
